rfcs/0132-meson-builds-nix.md

```diff
@@ -0,0 +1,135 @@
+---
+feature: meson_builds_nix
+start-date: 2022-08-25
+author: Anderson Torres
+co-authors: @p01arst0rm
+shepherd-team:
+shepherd-leader:
+related-issues:
+---
+
+# Summary
+[summary]: #summary
+
+Use meson as an alternative build system for Nix.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, the Nix evaluator and its companion toolset generated from the Nix source tree are built using the typical `./configure` shell script that relies on autoconf, along with standard GNU Make utility.
+
+Over time, this build system has been modified to keep up with the development needs of Nix project. However, it has reached a state where the build system became clunky and plastered, hard to understand and modify, consequently making improvements to the project as a whole very difficult.
+
+In particular, many changes have been introduced that impact compatibility outside Linux and NixOS niches. These issues can hinder development on other platforms, including but not limited to Unix-like systems.
+
+In light of this state of things, we propose a novel alternative to the current building infrastructure.
+
+We expect to accomplish, among other goals,
+
+- better code structuring and documentation;
+- improved cross-platform support, especially outside NixOS and Linux programming environments
+- shorter build times;
+- improved unit testing;
+- an overall improved user experience.
+
+# Detailed design
+[design]: #detailed-design
+
+A carefully crafted set of Meson files should be included in the Nix repository, providing a description on how to deploy the Nix repository, generating all the expected artifacts (command line tools, libraries, configuration files etc.)
+
+This novel building infrastructure should be able to provide at least feature parity with the current quasi-autotools implementation, albeit in a different user interface.
+
+# Examples and Interactions
+[examples-and-interactions]: #examples-and-interactions
+
+Currently @p01arst0rm is writing an implementation from scratch.
+
+Here is a table comparing some expected interactions:
+
+|----------------------|--------------------|------------------------------|
+| Action               | Current            | Meson                        |
+|----------------------|--------------------|------------------------------|
+| Configuring          | `./configure`      | `meson setup build_dir`      |
+| Building             | `make`             | `meson -C build_dir compile` |
+| Installing           | `make install`     | `meson -C build_dir install` |
+| Command-line Options | `--enable-gc=true` | `-Dgc=enabled`               |
```

Comment:
consider adding the uninstall command here too?

Reply:
Maybe, maybe not. uninstall is not a sure-fire uninstall because install can call arbitrary code.

Reply:
meson can also run arbitrary code on uninstall, and can set uninstall hooks